workspaces/adventure-pack/goodies/typescript/Iterator.prototype.drop/index.test.ts

```diff
@@ -0,0 +1,78 @@
+import { describe, expect, it } from "@jest/globals";
+
+import { iteratorPrototype } from "../Iterator.prototype";
+delete (iteratorPrototype as unknown as Record<string, unknown>).drop;
+
+import "./index";
+
+describe("Iterator.prototype.drop", () => {
+  it("drops the specified number of elements from the iterator", () => {
+    const result = Array.from([10, 22, 33, 47, 58].values().drop(3));
+    expect(result).toStrictEqual([47, 58]);
+  });
+
+  it("returns an empty iterator if the specified number is greater than the iterator length", () => {
+    const result = Array.from([10, 22, 33].values().drop(5));
+    expect(result).toStrictEqual([]);
+  });
+
+  it("returns the full iterator if the specified number is 0", () => {
+    const result = Array.from(["cat", "dog", "otter", "bear"].values().drop(0));
+    expect(result).toStrictEqual(["cat", "dog", "otter", "bear"]);
+  });
+
+  it("returns an empty iterator for an empty iterator", () => {
+    const result = Array.from([].values().drop(3));
+    expect(result).toStrictEqual([]);
+  });
+
+   it("drops the specified number of elements from an infinite iterator", () => {
+    function* infiniteNumbers() {
+      while (true) {
+        yield Math.floor(Math.random() * 100);
+      }
+    }
+
+    // Drop the first 2 elements, then take the next 5 to avoid infinte loop
+    const iterator = infiniteNumbers().drop(2).take(5);
+    const result = [];
+
+    for (const num of iterator) {
+      result.push(num);
+    }
+
+    expect(result.length).toBe(5);
+    expect(result.every(num => typeof num === 'number')).toBe(true);
+  });
+
+  it("throws when dropping elements from an iterator that errors before the drop limit", () => {
+    function* generateNumbers() {
+      yield 35;
+      throw new Error();
+    }
+
+    expect(() => {
+      Array.from(generateNumbers().drop(1));
+    }).toThrow();
+  });
+
+  it("doesn't reach errors in the underlying iterator that happen before the drop limit", () => {
+    function* generateNumbers() {
+      yield "Maine coon";
```

Comment:
The function is named `generateNumbers`, but this is not a number.

Reply:
meow